src/Compiler/NodeVisitor/ChangeFiltersNodeVisitor.php

```diff
@@ -238,4 +241,20 @@ private function createFilterCallNode(string $filterName, array $args): ?Node
             $args
         );
     }
+
+    /**
+     * @param ReflectionFunction|ReflectionMethod $reflection
+     * @param Arg[]|VariadicPlaceholder[] $args
+     * @return Arg[]|VariadicPlaceholder[]
+     */
+    private function updateArgs($reflection, array $args): array
+    {
+        $parameter = $reflection->getParameters()[0] ?? null;
+        if ($parameter instanceof ReflectionParameter && $parameter->getType() instanceof ReflectionNamedType && $parameter->getType()->getName() === FilterInfo::class) {
+            $args = array_merge([
```

Comment:
array_unshift would be cleaner

Reply:
This part is unchanged, for me array_merge is more clear. But I can change it to array_unshift

Reply:
If you like merge more then keep it